android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -468,11 +470,25 @@ public void onSizeChanged(int width, int height, int oldWidth, int oldHeight) {
     int bannerHeight = KMManager.getBannerHeight(context);
     int oskHeight = KMManager.getKeyboardHeight(context);
 
+    // If this condition is met, the current layout params are likely mismatched
+    // with the device's actual orientation.  Re-enforce & try again.
     if(bannerHeight + oskHeight != height) {
-      // We'll proceed, but cautiously and with logging.
-      KMLog.LogInfo(TAG, "Height mismatch: onSizeChanged = " + height + ", our version = " + (bannerHeight + oskHeight));
```

Comment:
As this log message seems _very_ strongly correlated with the current keyboard-height mismatch issue, it seems to be a very reasonable condition to utilize for carrying out corrective behavior.